src/gflownet/tasks/qm9/qm9.py

```diff
@@ -127,79 +129,73 @@ def compute_flat_rewards(self, mols: List[RDMol]) -> Tuple[FlatRewards, Tensor]:
         Z_params = list(model.logZ.parameters())
         non_Z_params = [i for i in self.model.parameters() if all(id(i) != id(j) for j in Z_params)]
         self.opt = torch.optim.Adam(
             non_Z_params,
-            hps["learning_rate"],
-            (hps["momentum"], 0.999),
-            weight_decay=hps["weight_decay"],
-            eps=hps["adam_eps"],
+            self.cfg.opt.learning_rate,
+            (self.cfg.opt.momentum, 0.999),
+            weight_decay=self.cfg.opt.weight_decay,
+            eps=self.cfg.opt.adam_eps,
+        )
+        self.opt_Z = torch.optim.Adam(Z_params, self.cfg.opt.learning_rate, (0.9, 0.999))
+        self.lr_sched = torch.optim.lr_scheduler.LambdaLR(self.opt, lambda steps: 2 ** (-steps / self.cfg.opt.lr_decay))
+        self.lr_sched_Z = torch.optim.lr_scheduler.LambdaLR(
+            self.opt_Z, lambda steps: 2 ** (-steps / self.cfg.opt.lr_decay)
         )
-        self.opt_Z = torch.optim.Adam(Z_params, hps["learning_rate"], (0.9, 0.999))
-        self.lr_sched = torch.optim.lr_scheduler.LambdaLR(self.opt, lambda steps: 2 ** (-steps / hps["lr_decay"]))
-        self.lr_sched_Z = torch.optim.lr_scheduler.LambdaLR(self.opt_Z, lambda steps: 2 ** (-steps / hps["Z_lr_decay"]))
 
-        self.sampling_tau = hps["sampling_tau"]
+        self.sampling_tau = self.cfg.algo.sampling_tau
         if self.sampling_tau > 0:
             self.sampling_model = copy.deepcopy(model)
         else:
             self.sampling_model = self.model
-        eps = hps["tb_epsilon"]
-        hps["tb_epsilon"] = ast.literal_eval(eps) if isinstance(eps, str) else eps
-        self.algo = TrajectoryBalance(self.env, self.ctx, self.rng, hps, max_nodes=9)
+        self.algo = TrajectoryBalance(self.env, self.ctx, self.rng, self.cfg)
 
         self.task = QM9GapTask(
             dataset=self.training_data,
-            temperature_distribution=hps["temperature_sample_dist"],
-            temperature_parameters=hps["temperature_dist_params"],
-            num_thermometer_dim=hps["num_thermometer_dim"],
+            temperature_distribution=self.cfg.task.qm9.temperature_sample_dist,
+            temperature_parameters=self.cfg.task.qm9.temperature_dist_params,
+            num_thermometer_dim=self.cfg.task.qm9.num_thermometer_dim,
             wrap_model=self._wrap_for_mp,
         )
-        self.mb_size = hps["global_batch_size"]
-        self.clip_grad_param = hps["clip_grad_param"]
+        self.mb_size = self.cfg.algo.global_batch_size
+        self.clip_grad_param = self.cfg.opt.clip_grad_param
         self.clip_grad_callback = {
             "value": (lambda params: torch.nn.utils.clip_grad_value_(params, self.clip_grad_param)),
             "norm": (lambda params: torch.nn.utils.clip_grad_norm_(params, self.clip_grad_param)),
             "none": (lambda x: None),
-        }[hps["clip_grad_type"]]
+        }[self.cfg.opt.clip_grad_type]
```

Comment:
consider breaking this into two statements.

Reply:
Why `(lambda x: None)` instead of `lambda x: None`?

Reply:
I think that was inherited from `yapf`, but we're using `black` now.